src/java.desktop/share/classes/java/awt/color/ColorSpace.java

```diff
@@ -108,7 +108,11 @@
      * The number of components in the color space.
      */
     private final int numComponents;
-    private transient String [] compName = null;
+
+    /**
+     * Lazy-initialized names of components in the color space.
+     */
+    private transient volatile String [] compName;
```

Comment:
The volatile keyword is the only one necessary to fix this bug, everything else is a cleanup.